statsmodels/censored/tests/test_robustros.py

```diff
@@ -0,0 +1,428 @@
+        known_cols = ['modeled', 'res', 'cen']
+        npt.assert_array_equal(self.ros.data.columns.tolist(), known_cols)
+
+    def test_debug_attr(self):
+        ntools.assert_true(hasattr(self.ros, 'debug'))
+        ntools.assert_true(isinstance(self.ros.data, pd.DataFrame))
+
+    def test_debug_cols(self):
+
+        npt.assert_array_equal(
+            sorted(self.ros.debug.columns.tolist()),
+            sorted(self.known_debug_cols)
+        )
+
+    def test_plotting_positions(self):
+        pp = np.round(np.array(self.ros.debug.plot_pos), 3)
+        npt.assert_array_almost_equal(
+            sorted(pp),
+            sorted(self.known_plot_pos),
+            decimal=3
+        )
+
+    def test_N_obs(self):
+        ntools.assert_true(hasattr(self.ros, 'N_obs'))
+        npt.assert_equal(self.ros.N_obs, self.data.shape[0])
+
+    def test_N_cen(self):
+        ntools.assert_true(hasattr(self.ros, 'N_cen'))
+        npt.assert_equal(self.ros.N_cen, self.data[self.data.cen].shape[0])
+
+    def test_cohn_attr(self):
+        ntools.assert_true(hasattr(self.ros, 'cohn'))
+        ntools.assert_true(isinstance(self.ros.cohn, pd.DataFrame))
+
+    def test_cohn_A(self):
+        npt.assert_array_equal(self.known_cohn_A,
+                                  self.ros.cohn['A'].values)
+
+    def test_cohn_B(self):
+        npt.assert_array_equal(self.known_cohn_B,
+                                  self.ros.cohn['B'].values)
+
+    def test_cohn_C(self):
+        npt.assert_array_equal(self.known_cohn_C,
+                                  self.ros.cohn['C'].values)
+
+    def test_cohn_PE(self):
+        npt.assert_array_almost_equal(
+            self.known_cohn_PE, self.ros.cohn['PE'], decimal=4
+        )
+
+    def test_MR_rosEstimator(self):
+        modeled = np.array(self.ros.data.modeled)
+        modeled.sort()
+
+        npt.assert_array_almost_equal(self.known_modeled, modeled, decimal=2)
+
+    def test_dup_index_error(self):
+        data = self.data.append(self.data)
+        npt.assert_raises(ValueError, RobustROSEstimator, data)
+
+    def test_non_dataframe_error(self):
+        data = self.data.values
+        npt.assert_raises(ValueError, RobustROSEstimator, data)
+
+    def test_plot_default(self):
+        ax = self.ros.plot()
+        ntools.assert_true(isinstance(ax, plt.Axes))
+
+    def test_plot_ylogFalse_withAx(self):
+        fig, ax = plt.subplots()
+        ax = self.ros.plot(ylog=False, ax=ax)
+
+    @ntools.raises(ValueError)
+    def test_nonnumeric(self):
+        N = 20
+        data = pd.DataFrame({
+            'res': ['A'] * N,
+            'cen': [False] * N
+        })
+        ros = RobustROSEstimator(data, result_col='res', censorship_col='cen')
+
+    def test_transform_in(self):
+        ntools.assert_equal(self.ros.transform_in, np.log)
+
+    def test_transform_out(self):
+        ntools.assert_equal(self.ros.transform_out, np.exp)
+
+
+
+class test_ROS_Helsel_Arsenic(_base_ROS_Mixin):
```

Comment:
pep-8 class names are capitalized and without underline `TestROSHelselArsenic`
base class with tests are usually called `CheckXxxMixin`, e.g.  `CheckROSMixin`